beacon-chain/attestation/service.go

```diff
@@ -0,0 +1,110 @@
+// Package attestation defines the life-cycle and status of single and aggregated attestation.
+package attestation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/prysmaticlabs/prysm/beacon-chain/types"
+	"github.com/sirupsen/logrus"
+)
+
+var log = logrus.WithField("prefix", "attestation")
+
+// Service represents a service that handles the internal
+// logic of managing aggregated attestation.
+type Service struct {
+	ctx           context.Context
+	cancel        context.CancelFunc
+	handler       *Handler
+	broadcastFeed *event.Feed
+	broadcastChan chan *types.Attestation
```

Comment:
This should be protobufs for both channels,

Reply:
Really? I see every service is using `types` for chan. Blockchain service for example:
```
type ChainService struct {
        ...
	incomingBlockFeed              *event.Feed
	incomingBlockChan              chan *types.Block
        ...
}
```

Reply:
My idea was beacause attestations are going to go from attester -> beacon-node -> proposer, it would constantly have to be changed from types to protobufs and inversely which would add additional overhead for us. But yeah if the other services are using types , then it should probably fine to use types also then , maybe I am just being a bit pedantic  :sweat_smile:

Reply:
One of the reasons I kept it as `types` in other services is that some of these things are also received via p2p and we are probably going to use simple serialize for p2p instead of protos. Best to plan ahead just in case.